src/transformState.js

```diff
@@ -5,7 +5,33 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        for (const key in action.extraData) {
+          state[key] = action.extraData[key];
+        }
+        break;
+      case 'removeProperties':
+        removeProperties(action, state);
+        break;
+      case 'clear':
+        for (const key1 in state) {
+          delete state[key1];
+        }
+        break;
+      default:
+        throw new Error(`Unknown action type ${action.type}.`);
+    }
+  }
+
+  return state;
+}
+
+function removeProperties(a, b) {
```

Comment:
a and b tells nothing

Great naming makes code more understandable, we should look at the names and know what this is about and what it is going to do 

So  something like 'object' and 'keys to remove' would be better here